.github/workflows/codeql.yml

```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "dev" ]
+  pull_request:
+    branches: [ "dev" ]
+  schedule:
+    - cron: "16 23 * * 3"
```

Comment:
Do we need to run this on a schedule? Or is running on PRs and merges to dev enough?

Reply:
The rationale for running on a schedule (in addition to PRs) is,

> Scanning code on a schedule informs you about the latest vulnerabilities and errors that GitHub, security researchers, and the community discover, even when developers aren't actively maintaining the repository.

[reference](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#configuring-frequency)

Since Terra UI has a lot of active development currently, running on a schedule probably won't buy us much but I also don't think it'd do much harm so I don't feel strongly either way.